### ISSUE TITLE
Unexpected behavior in GitLab CI with cache and symlinks

### ISSUE THREAD

**Issue by roquie** [amphp/file#44]
TL;DR
To fix issue create a new `1.0.1` version from master where unresolved symlink was deleted. And do not use any symlinks in composer packages, please.

Hi! I found a small bug related to symlinks in 1.0.0 version:
https://github.com/amphp/file/tree/80cfffa8238f4ccfe2bcf425bcc18fd3a8cb5eeb/travis

This folder contains symlink which can't be resolved where using GitLab CI cache for packing/unpacking dependencies into zip-archive. It breaks the vendor dependencies in the docker image and project can't be run successfully.

Container will be runs with error:
```
Fatal error: Could not check compatibility between Amp\Http\Server\Router::onStart(Amp\Http\Server\Server $server): Amp\Promise and Amp\Http\Server\ServerObserver::onStart(Amp\Http\Server\HttpServer $server): Amp\Promise, because class Amp\Http\Server\Server is not available in /app/vendor/amphp/http-server-router/src/Router.php on line 262
```

This error does not have any relations with `class_exists` in there files or how it function works. It just side effect.

Step by step:
1) Configure GitLab CI for two stages, first stage for install composer requirements and cache them, second stage for build docker image.
2) First stage runs `composer install --ignore-platform-reqs`, then cache `vendor/` with GitLab CI standard functionality. Zip archive will be created from `vendor/` directory.
3) On second stage, zip archive will be unzipped to git checkout directory.
4) CI throws warning:
```bash
Checking cache for composer-cache-master-2...
Downloading cache.zip from http://172.17.42.175/gitlab-runner-cache/gitlab_runner/project/1452/composer-cache-master-2 
WARNING: vendor/amphp/file/travis/install-uv.sh: chmod vendor/amphp/file/travis/install-uv.sh: no such file or directory (suppressing repeats) 
Successfully extracted cache
```
5) Docker image "successfully" created with broken files.
6) Container runs with error above.

GitLab CI issue: https://gitlab.com/gitlab-org/gitlab-runner/issues/3385

Please, release new version.

**Comment by roquie** [comment]
Maybe I incorrectly reported the problem... It really can be fixed with copy-paste of `install-uv.sh` instead of usage symlink and push to button to release a 1.0.1 version.

**Comment by roquie** [comment]
Please, don't use symbolic links in packages ...
https://github.com/sebastianbergmann/php-file-iterator/issues/67

**Comment by roquie** [comment]
https://github.com/amphp/windows-registry/issues/2

**Comment by trowski** [comment]
The `install-uv.sh` symlink was replaced some time ago with a file in this repo. If GitLab CI has issues with symlinks, please report this as a bug to GitLab.

**Comment by roquie** [comment]
> The install-uv.sh symlink was replaced some time ago with a file in this repo
Yeah. But it was be replaced in `master` and `1.0.0` version still contains symlinks. I would like what you released 1.0.1 version where it replaced with script.

GitLab has issue, but it's not resolved 2 years. So sad.

**Comment by trowski** [comment]
Tagged a `1.0.1`.
